sql-parser/src/test/java/io/crate/sql/IntervalLiteralTest.java

```diff
@@ -0,0 +1,114 @@
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.YEAR));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testMonth() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL +'1' MONTH");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.POSITIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.MONTH));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testDay() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL +'1' DAY");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.POSITIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.DAY));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testHour() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL +'1' HOUR");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.POSITIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.HOUR));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testMinute() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL +'1' MINUTE");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.POSITIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.MINUTE));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testSecond() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL +'1' SECOND");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.POSITIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.SECOND));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testNegative() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL -'1' HOUR");
+        assertThat(interval.getValue(), is("1"));
+        assertThat(interval.getSign(), is(IntervalLiteral.Sign.NEGATIVE));
+        assertThat(interval.getStartField(), is(IntervalLiteral.IntervalField.HOUR));
+        assertThat(interval.getEndField(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testTo() {
+        IntervalLiteral interval = (IntervalLiteral) SqlParser.createExpression("INTERVAL '1' HOUR TO SECOND");
```

Comment:
Are there any invalid `X TO Y` cases we should test / have to take care of?

Reply:
Well spotted, according to the https://crate.io/docs/sql-99/en/latest/chapters/08.html#interval-literal-s the start can not be less significant than the end. `INTERVAL '1' HOUR TO DAY` should not be valid.